bundles/statistics/statsgrid/handler/IndicatorFormHandler.js

```diff
@@ -0,0 +1,516 @@
+                datasets,
+                indicatorName: Oskari.getLocalized(ind.name),
+                indicatorDescription: Oskari.getLocalized(ind.description),
+                indicatorDatasource: Oskari.getLocalized(ind.source)
+            });
+        });
+    }
+
+    setIndicatorName (value) {
+        this.updateState({
+            indicatorName: value
+        });
+    }
+
+    setIndicatorDescription (value) {
+        this.updateState({
+            indicatorDescription: value
+        });
+    }
+
+    setIndicatorDatasource (value) {
+        this.updateState({
+            indicatorDatasource: value
+        });
+    }
+
+    setDatasetYear (value) {
+        this.updateState({
+            datasetYear: value
+        });
+    }
+
+    setDatasetRegionset (value) {
+        this.updateState({
+            datasetRegionset: value
+        });
+    }
+
+    setClipboardValue (value) {
+        this.updateState({
+            clipboardValue: value
+        });
+    }
+
+    addStatisticalData () {
+        if (this.state.datasetYear.length === 0 || isNaN(this.state.datasetYear)) {
+            Messaging.error(this.loc('errors.myIndicatorYearInput'));
+            return;
+        }
+        if (!this.state.datasetRegionset) {
+            Messaging.error(this.loc('errors.myIndicatorRegionselect'));
+            return;
+        }
+        const dataset = {
+            year: this.state.datasetYear,
+            regionset: this.state.datasetRegionset
+        };
+        this.showDataTable(this.state.datasource, dataset, this.state.indicator);
+    }
+
+    updateFormData (value, regionId) {
+        const regions = this.state.formData?.regions?.map(region => {
+            if (region.id === regionId) {
+                return {
+                    ...region,
+                    value: value
+                };
+            }
+            return region;
+        });
+        this.updateState({
+            formData: {
+                ...this.state.formData,
+                regions
+            }
+        });
+    }
+
+    showDataTable (datasourceId, dataset, indicatorId) {
+        this.updateState({
+            loading: true
+        });
+        const regionset = this.service.getRegionsets(dataset.regionset);
+        const labels = {};
+        labels[dataset.regionset] = regionset.name;
+        let formRegions;
+        const promise = new Promise((resolve, reject) => {
+            this.service.getRegions(regionset.id, (err, regions) => {
+                if (err) {
+                    reject(new Error(Oskari.getMsg('StatsGrid', 'errors.regionsDataError')));
+                    return;
+                }
+                const showDataForm = (regions, data) => {
+                    data = data || {};
+                    formRegions = [...regions].sort((a, b) => a.name.localeCompare(b.name)).map((region) => {
+                        // TODO: include existing values per region when editing existing dataset
```

Comment:
Is there something TODO/missing here still? Looks like showDataForm() is called with the data if we have an indicator id?

Reply:
Should be good to to go, TODO came with copy-paste from old code.